src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsTagger.cs

```diff
@@ -154,7 +155,8 @@ public IReadOnlyList<TagSpan<IntraTextAdornmentTag>> GetTags(NormalizedSnapshotS
                     // active view and disregards and requests for tags not in that view
                     var fullSpan = new SnapshotSpan(snapshot, 0, snapshot.Length);
                     var tags = _tagAggregator.GetTags(new NormalizedSnapshotSpanCollection(fullSpan));
-                    foreach (var tag in tags)
+                    var uniqueTags = tags.GroupBy(tag => tag.Tag.Hint.Span.Start).Select(group => group.First());
```

Comment:
isn't this just locally having a HashSet<int> and only processing a tag if it the first we see at a particular location?